modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotTraverse.scala

```diff
@@ -90,6 +92,12 @@ object GlobalSnapshotTraverse {
             case Right(globalSnapshot)           => GlobalSnapshotInfoV1.toGlobalSnapshotInfo(globalSnapshot.info).pure[F]
           }
 
+          _ <- firstInfo.stateProof.flatMap { proof =>
+            Applicative[F].whenA(proof =!= firstInc.stateProof) {
+              (new Exception(s"Snapshot info does not match the snapshot at ordinal=${firstInc.ordinal.show}")).raiseError[F, Unit]
+            }
```

Comment:
If you have applicative syntax imported then it can be simplified:

```scala
(new Exception(s"Snapshot info does not match the snapshot at ordinal=${firstInc.ordinal.show}"))
   .raiseError[F, Unit]
   .whenA(proof =!= firstInc.stateProof)
```

Reply:
:+1: